Lib/threading.py

```diff
@@ -1177,9 +1177,10 @@ def cancel(self):
         self.finished.set()
 
     def run(self):
-        self.finished.wait(self.interval)
-        if not self.finished.is_set():
-            self.function(*self.args, **self.kwargs)
+        """Continue execution after wait until function returns True"""
+        while(not self.finished.wait(self.interval)):
```

Comment:
Done. But: codecov/patch — 86.95% of diff hit (target 100%) - some red lines in  Lib/test/test_threading.py. 
I have no idea how to cover tests by tests =/